csrc/host_ir/executor.cpp

```diff
@@ -88,6 +89,45 @@ bool HostIrExecutor::isCompiled() const {
   return (bool)host_ir_container_;
 }
 
+namespace {
+// Validates the sizes and strides of the input and output tensors
+// against the tensorviews and posts the communication.
+c10::intrusive_ptr<c10d::Work> validateAndPostSingleCommunication(
+    Communication* communication,
+    DeviceIdxType my_device_index,
+    c10d::Backend* backend,
+    at::Tensor input_tensor,
+    at::Tensor output_tensor,
+    const ExpressionEvaluator& expr_eval) {
+  if (input_tensor.defined()) {
```

Comment:
Did you ever find the input tensor is undefined? You can verify that by making this a NVF_ERROR and then `!test`.

Reply:
Yes. I saw PipelineTests failures.
Undefined tensors are allowed in certain cases such as inputs on non-root in scatter.